src/hooks/useSnackbar/useSnackbar.tsx

```diff
@@ -55,9 +60,24 @@ export const SnackbarProvider: React.FC = ({ children }) => {
     return id
   }, [])
 
-  const closeSnackbar = (id: string) => {
-    setSnackbars(snackbars.filter((snackbar) => snackbar.id !== id))
-  }
+  const updateSnackbar = useCallback((id: string, opts: Omit<DisplaySnackbarArgs, 'id'>) => {
+    setSnackbars((currentSnackbars) => {
+      const newSnackbars = currentSnackbars.map((snackbar) => {
```

Comment:
you could make it much shorter. We especially don't need to create `newSnackbar` variable here.

Reply:
Maybe 
```tsx
  const updateSnackbar = useCallback((id: string, opts: Omit<DisplaySnackbarArgs, 'id'>) => {
    setSnackbars((currentSnackbars) =>
      currentSnackbars.map((snackbar) => (snackbar.id === id ? { ...snackbar, ...opts } : snackbar))
    )
  }, [])
``` 
?